automated-testing/unit-testing/tdd_example.md

```diff
@@ -0,0 +1,218 @@
+# Test-Driven Development Example
+
+With this method, rather than writing all your tests up front, you write one test a time and then switch to write the
+system code that would make that test pass. It's important to write the bare minimum of code necessary even if it is not
+actually "correct". Once the test passes you can refactor the code to make it maybe make more sense, but again the logic
```

Comment:
I think the "bare minimum" concept is best illustrated as trying to "cheat" your own test. 

Can you pass it with a simple "print" statement? Then write that print statement. Once you have something stupid that passes the test, improve the test so your cheat doesn't work anymore. Then go back and write the minimal code to pass it agani.

With this back and forth process, you only expand your code by small functional units, and your tests will rigorously push all the capabilities and limits of the target system.

Reply:
that is a great way of putting it! I will update to include that analogy